Added ability to set default min/magnification filtering

Super-Genius opened the pull request:
for 2D Textures and Font atlas.

The default min/magnification settings for rendering textures was GL_LINEAR, which blurred textures.  This allows the user to set the default filtering when creating textures.

Super-Genius committed:
Added ability to set default min/magnification filtering for 2D Textures and Atl

pandamicro commented:
@xiaofeng11 @liamcindy Please review this PR, @WenhaiLin please also confirm

Super-Genius committed:
Adds changes to FreeTypeFont to set the default DPI rendering.

xiaofeng11 commented:
Logic of these code is OK, but I'm not familiar with OpenGL programming, @liamcindy @WenhaiLin please confirm does these setting change will change the render effect of follow-up operations

WenhaiLin commented on cocos/2d/CCFontFreeType.cpp:
```diff
@@ -38,6 +39,8 @@ FT_Library FontFreeType::_FTlibrary;
 bool       FontFreeType::_FTInitialized = false;
 const int  FontFreeType::DistanceMapSpread = 3;
 
+int FontFreeType::_defaultFontDPI;
```
Please specify the initial value of `FontFreeType::_defaultFontDPI`.

Super-Genius committed:
Set default fontDPI settings

Commit:
Fixes FreeType font renderings (fonts are sampled * 64 in x direction for better FreeType Rendering to bitmap.

Commit:
These templates so that you can install a single copy of Cocos2D-x source code and use that directory when generating project files.

Commit:
Fixed horizontal kerning calculations for new Transform of font when rendering.

HStokes committed:
- Fix vcproj path reference from cocos2d-s to cocos2d-x

Commit:
Fixes for Mac build

HStokes committed:
Merge branch 'z4master' of https://github.com/Z4Gaming/cocos2d-x into z4master

Conflicts:
	.gitignore

HStokes committed:
Fixed merge conflict in .gitignore

dbanyaiGamingArts committed:
- Added a check in UICheckBox.cpp's onTouchEnded. The function now checks whether the click release is within the bounds of the check box.

Without this fix, you can click inside of a checkbox, drag to the outside of it, and release. This will trigger a bug where the checkbox will look checked but never go through the "check" functionality and will therefore appear to be in an incorrect state.

Super-Genius committed:
Merge pull request #2 from Z4Gaming/hotfix/checkboxfix

- Added a check in UICheckBox.cpp's onTouchEnded. The function now ch…

minggo commented:
This PR includes so many unneeded commits. So i will close it.

minggo commented:
@klhurley Could you please send a new PR?
